### PR TITLE
ModalMissionComplete Tests

### PR DESCRIPTION
**Pull request by danZzyy**
image

### CODE REVIEW

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 1]
```diff
@@ -112,17 +175,16 @@ describe("ModalMissionComplete", function () {
         neighborhood = new NeighborhoodMock();
         neighborhood.properties.name = "Test Neighborhood";
 
-        modalMissionMock = new ModalMissionMock(modalModel)
     });
 
-    describe("`show` method", function () {
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why the name change?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 2]
```diff
@@ -85,20 +147,21 @@ describe("ModalMissionComplete", function () {
         uiModalMissionComplete.obstacleCount = $uiModalMissionCompleteFixture.find("#modal-mission-complete-obstacle-count");
         uiModalMissionComplete.surfaceProblemCount = $uiModalMissionCompleteFixture.find("#modal-mission-complete-surface-problem-count");
         uiModalMissionComplete.otherCount = $uiModalMissionCompleteFixture.find("#modal-mission-complete-other-count");
+        this.uiModalMissionComplete = uiModalMissionComplete;
 
-        var modalModel = _.clone(Backbone.Events);
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why did you delete modalModel?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 3]
```diff
@@ -112,17 +175,16 @@ describe("ModalMissionComplete", function () {
...
-        it("should open a modal window", function () {
-            modalMissionComplete.hide();
+    describe("`hide` method", function () {
+        it("should hide a modal window", function () {
+            modal.hide();
             expect(uiModalMissionComplete.holder.css('visibility')).toBe('hidden');
             expect(uiModalMissionComplete.foreground.css('visibility')).toBe('hidden');
             expect(uiModalMissionComplete.background.css('visibility')).toBe('hidden');
 
-            modalMissionComplete.show(mission, neighborhood);
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why did you change the name from modalMissionComplete to modal?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 4]
```diff
@@ -131,73 +193,116 @@ describe("ModalMissionComplete", function () {
...
+    describe("setMissionTitle method", function () {
+        it("should change the text in the title html", function(){
+            expect(uiModalMissionComplete.missionTitle.html()).toBe('');
+            modal.setMissionTitle("Test Title");
+            expect(uiModalMissionComplete.missionTitle.html()).toBe("Test Title");
+            });
+    });
 
-    // Mocks
-    function MissionMock () {
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why did you delete the MissionMock? It wasn't used?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 5]
```diff
@@ -131,73 +193,116 @@ describe("ModalMissionComplete", function () {
...
+        });
+        it("should update label counts", function () {
+            modal.update(mission, neighborhood);
+            expect(uiModalMissionComplete.curbRampCount.html()).toBe('0');
+            expect(uiModalMissionComplete.noCurbRampCount.html()).toBe('0');
+            expect(uiModalMissionComplete.obstacleCount.html()).toBe('0');
+            expect(uiModalMissionComplete.surfaceProblemCount.html()).toBe('0');
+            expect(uiModalMissionComplete.otherCount.html()).toBe('0');
 
-    function NeighborhoodMock() {
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why delete the NeighborhoodMock()?

File: public/javascripts/SVLabel/spec/SpecRunner.html [hunk 6]
```diff
@@ -19,6 +19,8 @@
     <script type="text/javascript" src="../lib/jasmine-2.0.2/jasmine.js"></script>
     <script type="text/javascript" src="../lib/jasmine-2.0.2/jasmine-html.js"></script>
     <script type="text/javascript" src="../lib/jasmine-2.0.2/boot.js"></script>
+    <script src="../../lib/underscore.min.js"></script>
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SpecRunner.html]: Why import again?

File: public/javascripts/SVLabel/src/SVLabel/mission/Mission.js [hunk 7]
```diff
@@ -346,7 +346,9 @@ function Mission(parameters) {
             return getProperty("distance");
         }
     }
-
+    function setLabelCountsAtCompletion(labelCounts){
```
Comment (kotarohara) [inline public/javascripts/SVLabel/src/SVLabel/mission/Mission.js]: What is this method? Can you comment?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js [hunk 8]
```diff
@@ -1,73 +1,134 @@
 describe("ModalMissionComplete", function () {
-    var modalMissionComplete;
```
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: Why did you delete these?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 4 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: line 13

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 5 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: line 33

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 3 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: it was like that in you last spec file before you split the modules further. I changed it to modalMissionComplete now

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 8 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: modalMissionMock was only used in the commented out test you had on line 152

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 4 above)
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: What line 13?

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 5 above)
Comment (kotarohara) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: What line 33? Describe.

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 4 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: line 13 and line 33 of the new ModalMissionComplete-spec.js

**Comment by kotarohara** [comment]
I commented and I need you to address. I need you to address them before I can merge this PR.

File: public/javascripts/SVLabel/src/SVLabel/mission/Mission.js (hunk 7 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/src/SVLabel/mission/Mission.js]: I think this was sticking around from the older code. It's not used so I will get rid of it

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 1 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: I split this up into on for show and one for hide

File: public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js (hunk 5 above)
Comment (danZzyy) [inline public/javascripts/SVLabel/spec/SVLabel/modal/ModalMissionComplete-spec.js]: all Mock definitions were moved to the top of the spec file

**Comment by danZzyy** [comment]
I did 'git merge master' which resulted in all those commits. I checked that the tests pass and the app isnt broken
